src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -480,6 +480,89 @@ public String getCookie() {
             return getDisplayName();
         }
 
+        static String findThreadName(FlowNode flowNode){
+            ThreadNameAction tna = flowNode.getPersistentAction(ThreadNameAction.class);
+
+            if (tna != null) {
+                return tna.getThreadName();
+            }
+            return null;
+        }
+
+        static String findStageName(FlowNode flowNode){
+            LabelAction la = flowNode.getPersistentAction(LabelAction.class);
```

Comment:
Note that `LabelAction` is used more broadly than just for `stage`, so the method name might be misleading.